apps/fem/poisson3.sim

```diff
@@ -0,0 +1,188 @@
+%
+%  Strong form (poisson problem)
+%
+% - \nabla^2 u = f(x,y) in \Omega (domain)
+%            u = 0      on \Gamma (boundary)
+%
+%  solved using linear functions on a tetrahedral mesh
+%
+
+element Node
+    x    : vector[3](float); % position
+    edge : bool;             % true if part of the boundary
+    u    : float;            % solution
+    i    : int;              % index of node
```

Comment:
The thought was to not expose this since they are meant to be sets that we can freely reorganized. But since the values of the indices in files doesn't really matter, only that the right ones are connected by each tetrahedron, maybe it's fine to expose them in C++.

However, if the user loads a file and then write it back it may be that the index values have changed (but not the topology). Would that be acceptable behavior? We can always turn that off in debug mode.